### ISSUE TITLE
BIDS Specification Terms PDF Table Generator 

### ISSUE THREAD

**Issue by nqueder** [ohbm/hackathon2020#183]
**Guidelines**

*We are very excited to meet you at the 2020 OHBM Brainhack 🎉* *To submit a project, you need to be an attendee of the 2020 OHBM Brainhack. We ask you to register first over [here](http://www.humanbrainmapping.org/HackathonReg/). Thank you!*

*We have prepared a checklist to help with your project submission. Here is how to proceed:*
 1. *Before filling in any part, please submit this issue*
 2. *Check items in the checklist below as you go through them*
 3. *Once you are done (at least all 'required' items must be provided), please delete the "Guidelines" section add a comment saying 'hi @ohbm/project-monitors: My project is ready!'*

Thank you!

*After step 1 (issue submitted), we will assign a 'project monitor' to follow your submission. If at any time you need help or anything is unclear, please add a comment and ping your project monitor. Our team is here to help!*

----------------------------
## Project info

**Title**: BIDS Specification Terms PDF Table Generator 

**Project lead**: Nazek Queder @nqueder

**[Timezone](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#timezone)**: UTC-7

**Hub**: The Americas

**Description**: 

The project aims to create a user-friendly python tool allowing the BIDS group to select, search, or add BIDS specification terms and export a PDF table of the chosen BIDS term along with appropriate term properties and instructions on adding new terms. If the user chooses to add a new term, the tool will create and export a new BIDS specification term as a JSON-LD file and push it to our [terms](https://github.com/NIDM-Terms/terms/tree/master/terms/BIDS_Terms) repository where discussion surrounding the new term can be captured using git/github. Such an effort will provide the BIDS working group with a user-friendly tool to build tables for the BIDS documentation while ensuring they have not re-purposed an existing BIDS specification term giving it conflicting definitions.  For additional information on the background for this project see BIDS specification [issue 423](https://github.com/bids-standard/bids-specification/issues/423).

**Link to project**: https://github.com/nqueder/bids_terms_to_pdf_table

**Mattermost handle**: @nazek.queder

**Goals for the OHBM Brainhack**

* Create a Python-based tool allowing users to specify subsets of [BIDS specification terms](https://github.com/NIDM-Terms/terms/tree/master/terms/BIDS_Terms) and properties of those terms to include in a PDF-formatted table for inclusion into specifications.
* Create JSON-LD files for newly added BIDS specification terms.
* Creating a document with clear instructions for the user on how to use our tool.
* Create an automated way to push newly created JSON-LD files to the [Interlex](https://scicrunch.org/nidm-terms) information resource.

**Good first issues**:

**Skills**: 

* Python 
* Data query
* Git - any level
* BIDS - Brain Imaging Data Structure
* Knowledge about JSON and JSON-LD is recommended but not required
* Someone who knows how to develop good, tidy and clear documentation for developing clear and well structured user’s instructions
* Someone with good research and document analysis skills

**Chat channel**:  https://mattermost.brainhack.org/brainhack/channels/hbm_bids-terms-to-pdf

**Image for the OHBM brainhack website** 

## Submission checklist
*Once the issue is submitted, please check items in this list as you add under 'Additional project info'*

Please include the following above (all required):
-   [x] Link to your project: could be a code repository, a shared document, etc. See [here](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#link-to-project)
-   [x] Include your [Mattermost handle](https://mattermost.brainhack.org/) (i.e. your username). If you do not have an account, please [sign up here](https://mattermost.brainhack.org/signup_email).
-   [x] Goals for the OHBM Brainhack: describe what you want to achieve during this brainhack. See [here](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#goals).
-   [ ] Flesh out at least 2 "good first issues": those are tasks that do not require any prior knowledge about your project, could be defined as issues in a GitHub repository, or in a shared document, cf [here](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#onboarding-2-good-first-issues).
-   [x] Skills: list skills that would be particularly suitable for your project. We ask you to include at least one non-coding skill, cf. [here](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#onboarding-skills).
-   [x] Chat channel: A link to a chat channel that will be used during the OHBM Brainhack. This can be an existing channel or a new one. We recommend using the [Brainhack space on mattermost](https://mattermost.brainhack.org/), cf. [here](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#chat).
-   [x] Video channel: Please create a video channel that will be used during the OHBM Brainhack and share it in your chat channel above. This can be an existing channel or a new one. For instance a [jitsi meet](https://meet.jit.si/) room, cf. [here](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#video-calls).
-   [ ] Provide an image of your project for the OHBM brainhack website

You can also include information about (all optional):
-   [ ] Number of participants, cf. [here](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#participant-capacity)
-   [ ] Twitter-size summary of your project pitch, cf. [here](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#twitter-size-summary-of-your-project-pitch)
-   [ ] Set up a kanban board on your repository to better divide the work and keep track of things, cf [here](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#set-up-a-kanban-board)
-   [ ] Project snippet for the OHBM Brainhack website, cf. [here](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#project-snippet-for-the-ohbm-brainhack-website)

We would like to think about how you will credit and onboard new members to your project. We recommend reading references from [this section](https://github.com/ohbm/hackathon2020/blob/master/.github/ISSUE_TEMPLATE/handbooks/projects.md#credit-and-onboarding). If you'd like to share your thoughts with future project participants, you can include information about (recommended):
-   [ ] Specify how will you acknowledge contributions (e.g. listing members on a contributing page).
-   [ ] Provide links to onboarding documents if you have some.

**Comment by nqueder** [comment]
hi @ohbm/project-monitors: My project is ready!

**Comment by nqueder** [comment]
hi @ohbm/project-monitors My project is ready!

**Comment by Remi-Gau** [comment]
hey @nqueder :wave: 

This looks good. :sparkles: 

I think that our @ohbm/hack2020-monitors will have another look at that soon.

Already added some labels. Check through our list of issue labels if some other ones could be added.

Do you have an image for the [website](https://ohbm.github.io/hackathon2020/hackathon/) we could use? 

Maybe the BIDS logo?

**Comment by nqueder** [comment]
Thanks! @Remi-Gau

An image for the website is attached below.

![BIDS-Terms](https://user-images.githubusercontent.com/47163431/84584825-e32ca700-adbd-11ea-8901-ffef608fa7cc.png)

**Comment by Remi-Gau** [comment]
Fantastic. 

@DorienHuijser Can I let you handle this one? :wink:

**Comment by jhlegarreta** [comment]
@nqueder  do you have a video channel for your project? Would you add that information to **the mattermost channel**, please? Thanks.

**Comment by nqueder** [comment]
@jhlegarreta of course! The link to my video channel has been added to the [mattermost channel](https://mattermost.brainhack.org/brainhack/channels/hbm_bids-terms-to-pdf) description.

**Comment by complexbrains** [comment]
Dear @nqueder would you mind adding 2 good first issues to your project and letting us know please. Once you are done with that, I will complete the project revision and your project will be ready to go! Thank you 🤗

**Comment by nqueder** [comment]
Hi @complexbrains !
My issue has been updated with two "good first issues" :)

**Comment by complexbrains** [comment]
Dear @nqueder Thank you so much for doing it so! Your project seems ready to go then! Welcome aboard! :hugs: 🎊
